polyfill/lib/calendar.mjs

```diff
@@ -323,13 +323,13 @@ function resolveNonLunisolarMonth(calendarDate, overflow = undefined, monthsPerY
     // rely on this function to constrain/reject out-of-range `month` values.
     if (overflow === 'reject') ES.RejectToRange(month, 1, monthsPerYear);
     if (overflow === 'constrain') month = ES.ConstrainToRange(month, 1, monthsPerYear);
-    monthCode = `M${month}`;
+    monthCode = `M${month.toString().padStart(2, '0')}`;
```

Comment:
Given how often this pattern is repeated, seems worthy of pulling out into a `buildMonthCode` function either locally in this file or as an `ES` method.  Would shave some bytes off the polyfill.

Reply:
done!